### DISCUSSION TITLE
What is the active image?

### DISCUSSION THREAD

**Discussion by GiulioDallaVecchia** [mcu-tools/mcuboot#1700] [Q&A]
Hi all,

I'm working on a project where I've set loading mode from RAM. The final application is copied to external RAM and run. I've set two slot on flash: primary and secondary slot. How can I know in which slot there is the active image? I need to know that in order to copy the newest application image to the correct slot.

**Comment by de-nordic** [comment]
Depends on configuration. With swap type configuration MCUboot boots primary slot. In case when DirectXIP is used then the slot with newer version will be executed, or the primary slot if both have the same version.

**Comment by GiulioDallaVecchia** [comment]
Hi @de-nordic , 

I'm not using DirectXIP mode, but RAMLOAD mode. With this mode the newer image will be executed. I need to know where is the newer image (slot 1 or 2?) in order to understand what is the slot that I can erase when I have to upload a new image.

**Comment by de-nordic** [comment]
I guess you are trying to figure this out from you app, not MCUboot.
So you basically have to use the bootutil functions to read headers from images and figure this way.
I am not that familiar with RAM load so I do not know how much it does differ from swap/direct-xip things, but you may also have to check the trailer flags to figure out which image is currently running, or you can compare image version, as gotten by the bootutil function to the image version the running app knows to have.

**Comment by GiulioDallaVecchia** [comment]
But I can to do only with the `boot_image_load_header(const struct flash_area *fa_p, struct image_header *hdr)` function, is it correct?

This function is not part of the last version v1.10.0

**Comment by de-nordic** [comment]
You are doing modification of MCUboot or the code will be invoked from something that MCUboot has already loaded to RAM?

**Comment by GiulioDallaVecchia** [comment]
When I have to upload a new application image, I need to know which slot has the oldest image, in order to delete and download the new image in this slot. To do that I need to call a MCUboot function to know which slot contain the oldest image that I can delete. Deleting the oldest image, if something happens during the updating, MCUboot can load older application version that is in the other slot.

If you use MCUBOOT_SWAP_USING_SCRATCH mode, the new image is always downloaded to the secondary slot and the swap with the primary slot (so that if something happens during the updating, MCUboot can reload older application version). If you use the MCUBOOT_RAM_LOAD mode (that is the mode that I have to use), there isn't any swapping mechanism, so I need to select manually in which slot I have to download the new image.

**Reply by de-nordic** [reply]
Then use the load header to reach headers for both slots and compare it against the one you store in your image in ram.
